dot/network/service.go

```diff
@@ -379,6 +383,54 @@ func (s *Service) SendMessage(msg NotificationsMessage) {
 	s.host.broadcast(msg)
 }
 
+func (s *Service) handleFinalityStream(stream libp2pnetwork.Stream) {
+	conn := stream.Conn()
+	if conn == nil {
+		logger.Error("Failed to get connection from stream")
+		return
+	}
+
+	peer := conn.RemotePeer()
+	s.readStream(stream, peer, s.decodeFinalityProofMessage, s.handleFinalityProofMessage)
+}
+
+func (s *Service) decodeFinalityProofMessage(in []byte, peer peer.ID) (Message, error) {
+	s.finalityProofRequestMu.RLock()
+	defer s.finalityProofRequestMu.RUnlock()
+
+	// Check if we are the requester
+	if _, requested := s.finalityProofRequest[peer]; requested {
+		// We requested, hence expect a response.
+		msg := new(FinalityProofResponse)
+		err := msg.Decode(in)
+		return msg, err
+	}
+
+	// We expecting a request at this point.
+	msg := new(FinalityProofRequest)
+	err := msg.Decode(in)
+	return msg, err
+}
+
+func (s *Service) handleFinalityProofMessage(peer peer.ID, msg Message) error {
+	req, ok := msg.(*FinalityProofRequest)
+	if !ok {
+		logger.Error("expected FinalityRequest from peer %s, got %+v", peer, msg)
+		return nil
+	}
+
+	// TODO: handle finality-proof request.
+	logger.Info("FinalityProof message %s", req)
+
+	resp := FinalityProofResponse{}
+	if err := s.host.send(peer, finalityProofID, &resp); err != nil {
+		logger.Error("failed to send FinalityProofResponse message", "peer", peer, "err", err)
+		return err
+	}
+
+	return nil
```

Comment:
could you update this to check whether it's `FinalityProofRequest` or `FinalityProofResponse` and log it accordingly? you don't need to handle anything at the moment